pkg/liqo-controller-manager/ip-controller/ip_controller.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2019-2023 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipctrl
+
+import (
+	"context"
+
+	"slices"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ipamv1alpha1 "github.com/liqotech/liqo/apis/ipam/v1alpha1"
+	"github.com/liqotech/liqo/apis/virtualkubelet/v1alpha1"
+	"github.com/liqotech/liqo/pkg/liqonet/ipam"
+	"github.com/liqotech/liqo/pkg/utils/getters"
+)
+
+const (
+	ipamIPFinalizer = "ip.ipam.liqo.io"
+)
+
+// IPReconciler reconciles a IP object.
+type IPReconciler struct {
+	client.Client
+	Scheme     *runtime.Scheme
+	IpamClient ipam.IpamClient
+}
+
+// +kubebuilder:rbac:groups=ipam.liqo.io,resources=ips,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ipam.liqo.io,resources=ips/status,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ipam.liqo.io,resources=ips/finalizers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=virtualkubelet.liqo.io,resources=virtualnodes,verbs=get;list;watch
+
+// Reconcile Ip objects.
+func (r *IPReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("Reconcilg IP %q", req.NamespacedName) // TODO:: delete
+	var ip ipamv1alpha1.IP
+	var desiredIP string
+
+	// Fetch the IP instance
+	if err := r.Get(ctx, req.NamespacedName, &ip); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof(" %q not found", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		klog.Errorf("an error occurred while getting IP %q: %v", req.NamespacedName, err)
+		return ctrl.Result{}, err
+	}
+
+	desiredIP = ip.Spec.IP
+
+	// Get the clusterIDs of all remote clusters
+	var virtualNodes v1alpha1.VirtualNodeList
+	if err := r.List(ctx, &virtualNodes); err != nil {
+		klog.Errorf("error while listing virtual nodes: %v", err)
+		return ctrl.Result{}, err
+	}
```

Comment:
https://github.com/fra98/liqo/blob/5f3d7e95abf3105666631df6a559e3f7f8ef3bb1/pkg/utils/getters/k8sGetters.go#L255-L260

Uses this package if you can. With labels.Everything() from "k8s.io/apimachinery/pkg/labels" you can get the same behaviour